### ISSUE TITLE
Cache is not working

### ISSUE THREAD

**Issue by Sumankhan420** [amitshekhariitbhu/Fast-Android-Networking#391]
I am using the following code in Android : 
`AndroidNetworking.post(url)
                .addQueryParameter("id","123")
                .setPriority(Priority.LOW)
                .getResponseOnlyIfCached()
                .build()
                .getAsJSONObject(new JSONObjectRequestListener() {
                    @Override
                    public void onResponse(JSONObject response) {
                        Log.e(TAG, "onResponse: "+response);
                    }

                    @Override
                    public void onError(ANError anError) {
                        Log.e(TAG, "onError: "+anError.getErrorDetail() );
                        Log.e(TAG, "onError: "+anError.getErrorCode() );
                    }
                });`

My URL Links to a PHP Webservice API which is working perfectly fine on browsers and IOS Platforms. 
There I have provided the following cache-controls : 

`header('Cache-Control: max-age=300, private');
header('Expires: Mon, 26 Jul 2019 05:00:00 GMT');
header('Content-type: application/json');

print_r($json);`

In Android I am getting the error : 

**onError: responseFromServerError
onError: 504**

If I remove the line "getResponseOnlyIfCached()" then the API works perfectly fine, but using Cache is not working. 

Thanks

**Comment by Sumankhan420** [comment]
yes

**Comment by amitshekhariitbhu** [comment]
Have you initialize like this `AndroidNetworking.initialize(getApplicationContext());`

**Comment by Sumankhan420** [comment]
Yes.that is exactly how I have initialised the library. I am having no
difficulty in using this library. I am using this library throughout my
project..

My only difficulty is in getting the cache to work whenever I try to apply
your cache code as shown above, then the error comes.

**Comment by Sumankhan420** [comment]
any update sir?

**Comment by blessonabraham** [comment]
@Sumankhan420  Hello, i know its a late answer, but i came across the same issue and found that Cache will only work for GET requests with most of the Android Networking Libraries!
